src/misc.ts

```diff
@@ -17,6 +17,15 @@ export type Mutable<
   [Key in keyof Omit<ObjectValue, TargetKey>]: ObjectValue[Key];
 };
 
+/**
+ * Get a type representing the public interface of the given type. The
+ * returned type will have all public properties, but will omit private
+ * properties.
+ *
+ * @template Interface - The interface to return a public representation of.
+ */
+export type PublicInterface<Interface> = Pick<Interface, keyof Interface>;
```

Comment:
My only concern with this is the name. It's a little weird to call this "PublicInterface" when it's a type, not an interface.

I meant "interface" in the English word sense, not like a TypeScript interface, but maybe there's a better name that can avoid this confusion.

I couldn't think of one though, and this seems good enough

Reply:
I guess we could call it `PublicType` 🤔 That doesn't have the same connotation as "public interface" though. Probably best as-is.